src/TypeScript/Analyzer/JavaScriptAnalyzer.cs

```diff
@@ -36,29 +36,36 @@
 namespace SonarLint.VisualStudio.TypeScript.Analyzer
 {
     [Export(typeof(IAnalyzer))]
-    internal class JavaScriptAnalyzer : IAnalyzer
+    internal sealed class JavaScriptAnalyzer : IAnalyzer, IDisposable
     {
-        private readonly IEslintBridgeClient eslintBridgeClient;
+        private readonly IEslintBridgeClientFactory eslintBridgeClientFactory;
+        private readonly IEslintBridgeProcess eslintBridgeProcess;
         private readonly IEslintBridgeIssueConverter issuesConverter;
         private readonly ILogger logger;
 
+        private IEslintBridgeClient javascriptClient;
+        private int javascriptServerPort;
+
         [ImportingConstructor]
-        public JavaScriptAnalyzer(IEslintBridgeClient eslintBridgeClient,
+        public JavaScriptAnalyzer(IEslintBridgeClientFactory eslintBridgeClientFactory,
+            IEslintBridgeProcess eslintBridgeProcess,
```

Comment:
IEslintBridgeProcess is still a singleton, I will change that in the next PR and create IEslintBridgeProcessFactory that it will call and hold reference to.

Reply:
What's the purpose of `IEslintBridgeProcessFactory` i.e. what would call it and when?
Currently, `EslintBridgeProcess` handles starting a new process if the previous one exited. What does the factory add?

Reply:
A new instance for `EslintBridgeProcess`, for typescript. On second thought, I will do it when we get to typescript.